src/main/java/ch/njol/skript/conditions/CondHasBuildRestrictions.java

```diff
@@ -0,0 +1,93 @@
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.aliases.ItemType;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import com.destroystokyo.paper.Namespaced;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.HashSet;
+import java.util.Set;
+
+@Name("Has Build Restrictions")
+@Description("Check if an item has any build restrictions.")
+@Examples({"player's tool has destroyable blocks",
+	"{_item} doesn't have placeable keys",
+	"{_item} is able to break stone and dirt",
+	"{_item} can not be placed on diamond ore"})
+@Since("INSERT VERSION")
+@RequiredPlugins("Paper")
+public class CondHasBuildRestrictions extends Condition {
+	static {
+		if (Skript.methodExists(ItemMeta.class, "hasDestroyableKeys")) {
+			Skript.registerCondition(CondHasBuildRestrictions.class,
+				"%itemtypes% (can|is able to) (break|destroy|mine|place:be placed on) %itemtypes%",
+				"%itemtypes% (can't|can[ ]not|is unable to) (break|destroy|mine|place:be placed on) %itemtypes%",
+				"%itemtypes% (has|have) [a|any] (destroy[able]|place:place[able| on]) (key|item|block)[s]",
+				"%itemtypes% (doesn't|does not|do not|don't) have [a|any] (destroy[able]|place:place[able| on]) (key|item|block)[s]",
```

Comment:
These two are quite awkward:
```
if {_item} has any place block 
if {_item} has a destroyable item # this one could get confusing
if {_item} does not have a placeable item # sounds like the "player has 1 stone" condition, not something that checks for restrictions.
```
Why not something like ``%itemtypes% (has|have) [a|any] (:break|:build) restrictions``?